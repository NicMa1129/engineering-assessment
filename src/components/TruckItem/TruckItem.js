import React from 'react';
import PropTypes from 'prop-types';
import './TruckItem.less';

const TruckItem = ({ locationDes, foodItems }) => {
  return (
    <div className="truckItem">
      <p className="location">location: {locationDes}</p>
      <p className="food">foods: {foodItems}</p>
    </div>
  );
};

TruckItem.defaultProps = {
  locationDes: '',
  foodItems: '',
};

TruckItem.propTypes = {
  locationDes: PropTypes.string,
  foodItems: PropTypes.string,
};

export default TruckItem;
