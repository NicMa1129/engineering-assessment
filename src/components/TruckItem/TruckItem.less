.truckItem {
  padding: var(--spacer-m);
  margin-bottom: var(--spacer-m);
  background-color: var(--color-off-white);
  
  & .location {
    margin-bottom: var(--spacer-s);
    font-size: var(--font-size-14);
    color: var(--color-dark);
  }

  & .food {
    font-size: var(--font-size-13);
    color: var(--color-gery);
  }
}