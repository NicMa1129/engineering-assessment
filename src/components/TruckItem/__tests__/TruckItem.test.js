import React from 'react';
import { render } from '@testing-library/react';
import TruckItem from '../TruckItem';

describe('<TruckItem />', () => {
  it('should render correctly', () => {
    const { container } = render(
      <TruckItem
        locationDes="MARKET ST: DRUMM ST intersection"
        foodItems="Fried Chicken: Fried Fish: Greens: Mac & Cheese: Peach Cobbler: and String beans"
      />
    );

    expect(container).toMatchSnapshot();
  });
});
