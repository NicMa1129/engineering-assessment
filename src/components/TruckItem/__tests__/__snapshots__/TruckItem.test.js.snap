// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TruckItem /> should render correctly 1`] = `
<div>
  <div
    class="truckItem"
  >
    <p
      class="location"
    >
      location: 
      MARKET ST: DRUMM ST intersection
    </p>
    <p
      class="food"
    >
      foods: 
      Fried Chicken: Fried Fish: Greens: Mac & Cheese: Peach Cobbler: and String beans
    </p>
  </div>
</div>
`;
